compose.py

```diff
@@ -0,0 +1,276 @@
+
+
+### Create container with docker build
+def build_docker_image(ddir, dockerfile_name, container_name):
+    p = subprocess.Popen(['docker', 'build', '-t', container_name, '-f', \
+        os.path.join(ddir, dockerfile_name), '.'])
+    p.wait()
+    fail_if(p.returncode != 0, 'docker build {} failed'.format(container_name))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    group_qv = parser.add_mutually_exclusive_group()
+    group_qv.add_argument('-q',
+                          '--quiet',
+                          action="store_true",
+                          required=False,
+                          help='Disable console output.')
+    group_qv.add_argument('-v',
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument('--output-name',
+                        type=str,
+                        required=False,
+                        help='Name for the generated Docker image. Default is "tritonserver".')
+    parser.add_argument(
+        '--work-dir',
+        type=str,
+        required=False,
+        help=
+        'Generated dockerfiles are placed here. Default to current directory.'
+    )
+    parser.add_argument(
+        '--version',
+        type=str,
+        required=False,
+        help=
+        'The Triton version. If not specified defaults to the value in the TRITON_VERSION file.'
+    )
+    parser.add_argument(
+        '--container-version',
+        type=str,
+        required=False,
+        help=
+        'The version to use for the generated Docker image. If not specified the container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--upstream-container-version',
+        type=str,
+        required=False,
+        help=
+        'The version of the NGC Docker images to use as a base for this image. If not specified the upstream container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--enable-gpu',
+        action="store_true",
+        required=False,
+        help='Generate a Triton image that supports GPU'
+    )
+    parser.add_argument(
+        '--backend',
+        action='append',
+        required=False,
+        help=
+        'Include <backend-name> in the generated Docker image. The flag may be specified multiple times.'
+    )
+    parser.add_argument(
+        '--repoagent',
+        action='append',
+        required=False,
+        help='Include <repoagent-name> in the generated Docker image. The flag may be specified multiple times.')
+    FLAGS = parser.parse_args()
+    
+    if FLAGS.work_dir is None:
+        FLAGS.work_dir = "."
+    if FLAGS.output_name is None:
+        FLAGS.output_name = "tritonserver"
+    # Determine the versions. Start with Triton version, if --version
```

Comment:
You only need a single version flag, --container-version, from which you determine the min and server container to pull from NGC. All the other version and build information you need to extract like you do for the build information below.